### DISCUSSION TITLE
[pure function] confluence

### DISCUSSION THREAD

**Discussion by dannypsnl** [FizzyElt/functional-programming#57] [General]
Confluence means that the order of performing various subcomputations doesn’t matter so long as they all finish: in the expression `(2 + 5) * (3 + 6)`, it doesn’t matter which addition you do first or whether you distribute the expressions over each other; the answer is the same.

The example of confluence in lambda calculus is **beta reduction**, and this is the most pure function similar formal idea I can find

**Comment by dannypsnl** [comment]
純函數跟特定系統連結太深了，就算直觀上兩個好像一樣，但在不同系統裡面可能一個純一個不純啊

舉個例子，在 C 語言裡面總是能用堆疊溢位影響任意函數的操作，這樣真的還有「純函數」嗎？實際上編譯器也是直接放任不管，出包是開發者的責任

**Reply by FizzyElt** [reply]
提兩個不同情境的案例，並請讀者自己留意有這件事

**Comment by dannypsnl** [comment]
https://types.pl/@pigworker/111320767229027862
